pkg/services/authz/rbac/service.go

```diff
@@ -97,9 +98,10 @@ func NewService(
 		idCache:         newCacheWrap[store.UserIdentifiers](cache, logger, tracer, longCacheTTL),
 		permCache:       newCacheWrap[map[string]bool](cache, logger, tracer, settings.CacheTTL),
 		permDenialCache: newCacheWrap[bool](cache, logger, tracer, settings.CacheTTL),
-		teamCache:       newCacheWrap[[]int64](cache, logger, tracer, settings.CacheTTL),
+		userTeamCache:   newCacheWrap[[]int64](cache, logger, tracer, settings.CacheTTL),
 		basicRoleCache:  newCacheWrap[store.BasicRole](cache, logger, tracer, settings.CacheTTL),
 		folderCache:     newCacheWrap[folderTree](cache, logger, tracer, settings.CacheTTL),
+		teamIDCache:     newCacheWrap[map[int64]string](cache, logger, tracer, settings.CacheTTL),
```

Comment:
Would it make sense to use the `longCacheTTL` here? It's currently not configurable, which is not great (hardcoded to 2 mins). But I feel like it makes sense for team IDs and user IDs to have the same cache TTL, as IMO they're similarly unchanging (unless someone creates a new team, but you deal with that case separately anyway).